### PR TITLE
astro 1.32.0

### PR DESCRIPTION
**Pull request by neel-astro**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  ==> echo 'y' | /opt/homebrew/Cellar/astro/1.32.0/bin/astro dev init
  Error: astro: failed
  An exception occurred within a child process:
    Minitest::Assertion: Expected /^Initializing Astro project*/ to match "/private/tmp/astro-test-20241217-8888-tgdobt is not an empty directory. Are you sure you want to initialize a project here? (y/n) Initialized empty Astro project in /private/tmp/astro-test-20241217-8888-tgdobt\n".
```

**Comment by neel-astro** [comment]
> ```
>   ==> echo 'y' | /opt/homebrew/Cellar/astro/1.32.0/bin/astro dev init
>   Error: astro: failed
>   An exception occurred within a child process:
>     Minitest::Assertion: Expected /^Initializing Astro project*/ to match "/private/tmp/astro-test-20241217-8888-tgdobt is not an empty directory. Are you sure you want to initialize a project here? (y/n) Initialized empty Astro project in /private/tmp/astro-test-20241217-8888-tgdobt\n".
> ```

Thanks @chenrui333 for the prompt, there had been some adjustments in the new release, which I failed to incorporate earlier. I have adjusted the tests now 🙂.

File: Formula/a/astro.rb [hunk 1]
```diff
@@ -32,8 +33,8 @@ def install
     version_output = shell_output("#{bin}/astro version")
     assert_match("Astro CLI Version: #{version}", version_output)
 
-    run_output = shell_output("echo 'y' | #{bin}/astro dev init")
-    assert_match(/^Initializing Astro project*/, run_output)
+    run_output = shell_output("mkdir /tmp/astro-project && cd /tmp/astro-project && #{bin}/astro dev init")
```
Comment (chenrui333) [inline Formula/a/astro.rb:36]: would this be good enough?

```suggestion
    run_output = shell_output("#{bin}/astro dev init")
```

Comment (neel-astro) [reply Formula/a/astro.rb:36]: Astro CLI docs for reference around the actual usage: https://www.astronomer.io/docs/astro/cli/develop-project#create-an-astro-project

Comment (chenrui333) [reply Formula/a/astro.rb:36]: can it be configured to use `testpath` rather than `tmp` then?

Comment (neel-astro) [reply Formula/a/astro.rb:36]: good callout, have updated the test to use `testpath`

File: Formula/a/astro.rb [hunk 2]
```diff
@@ -32,9 +35,12 @@ def install
     version_output = shell_output("#{bin}/astro version")
     assert_match("Astro CLI Version: #{version}", version_output)
 
-    run_output = shell_output("echo 'y' | #{bin}/astro dev init")
-    assert_match(/^Initializing Astro project*/, run_output)
-    assert_predicate testpath/".astro/config.yaml", :exist?
+    mkdir testpath/"astro-project"
+    cd testpath/"astro-project" do
+      run_output = shell_output("#{bin}/astro dev init")
+      assert_match(/^Initialized empty Astro project*/, run_output)
```
Comment (chenrui333) [inline Formula/a/astro.rb:41]: does not have to be regex call
```suggestion
      assert_match "Initialized empty Astro project", run_output
```

File: Formula/a/astro.rb [hunk 3]
```diff
@@ -32,9 +35,12 @@ def install
...
+      assert_predicate testpath/".astro/config.yaml", :exist?
```
Comment (chenrui333) [inline Formula/a/astro.rb:42]: better syntax :)
```suggestion
      assert_path_exists testpath/".astro/config.yaml"
```

File: Formula/a/astro.rb [hunk 4]
```diff
@@ -20,6 +20,9 @@ class Astro < Formula
   end
 
   depends_on "go" => :build
+  on_macos do
+    depends_on "podman"
+  end
```
Comment (chenrui333) [inline Formula/a/astro.rb:25]: it is only for macos?

Comment (neel-astro) [reply Formula/a/astro.rb:25]: Yes, podman dependency is only limited to macos

File: Formula/a/astro.rb (hunk 3 above)
Comment (neel-astro) [reply Formula/a/astro.rb]: thanks for the recommendation, updated the logic

File: Formula/a/astro.rb (hunk 2 above)
Comment (neel-astro) [reply Formula/a/astro.rb]: thanks for the recommendation, updated the logic

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**